Fix uncalibrated RC input being used

MaEtUgR opened the pull request:
<!--

Thank you for your contribution!

Get early feedback through
- Dronecode Discord: https://discord.gg/dronecode
- PX4 Discuss: http://discuss.px4.io/
- opening a draft pr and sharing the link

-->

### Solved Problem
When testing for https://github.com/PX4/PX4-Autopilot/issues/21506 I often reset all parameters to go through a fresh setup and found that my RC input data from a crossfire receiver was used even though I didn't calibrate it yet. The values were all static zero because it was uncalibrated but still assuming to have pilot input even though there is none is a safety concern.

### Solution
- (Refactor topic object naming in RCInput.)
- Start using the `valid` flag in the `manual_control_input` topic to signal that the input is not actually calibrated and shouldn't be considered valid and used. The RC is considered calibrated if the channel count parameter `RC_CHAN_CNT` is set and at least one of the stick axes roll, pitch, yaw or throttle is mapped to an RC channel. Further the input is only considered valid if the number of channels in it actually matches what was calibrated (`RC_CHAN_CNT`).

### Changelog Entry
For release notes:
```
Bugfix Uncalibrated remote input is now considered invalid
```

### Alternatives
What's not ideal in my eyes is the reporting to the user. He cannot screw up anymore by accidentally using uncalibrated stick input but he's not informed about the details about why the stick input does not get used. QGC shows a red remote icon but feedback could be better. Should I duplicate all these checks into the RC calibration check here? https://github.com/PX4/PX4-Autopilot/blob/1d4b66fcbc103eb2e81eb708d863595679989884/src/modules/commander/HealthAndArmingChecks/checks/rcCalibrationCheck.cpp#L81 I considered that but then we need to change the workflow of default RC configuration because it will immediately fail and you will not be able to fly without RC without disabling it. I think this is the way to go but I don't feel comfortable introducing this change for 1.14 since we don't have UI changes and RC disabled by default to go with it yet.

### Test coverage
I did bench testing using the same setup that I found the issue with. On a fresh setup the `manual_control_input` is published with the uncalibrated values but flagged as invalid and therefore does not get selected/used.
<img src="https://user-images.githubusercontent.com/4668506/235889619-a3b5c50d-3d84-437d-a6d2-54e7c6189739.png" width="350">
Once the RC is successfully calibrated it gets flagged valid and is normally used.
<img src="https://user-images.githubusercontent.com/4668506/235889949-ac6ff101-515f-4a68-895a-1cdc77681f81.png" width="350">

bkueng commented on src/modules/rc_update/rc_update.cpp:
```diff
@@ -687,6 +692,7 @@ void RCUpdate::UpdateManualControlInput(const hrt_abstime &timestamp_sample)
 	manual_control_input.aux4  = get_rc_value(rc_channels_s::FUNCTION_AUX_4,   -1.f, 1.f);
 	manual_control_input.aux5  = get_rc_value(rc_channels_s::FUNCTION_AUX_5,   -1.f, 1.f);
 	manual_control_input.aux6  = get_rc_value(rc_channels_s::FUNCTION_AUX_6,   -1.f, 1.f);
+	manual_control_input.valid = _rc_calibrated && (_param_rc_chan_cnt.get() == _channel_count_previous);
```
I'm not sure about this `_param_rc_chan_cnt.get() == _channel_count_previous`. If either the param changes or the count for some reason in flight, you lose RC.

MaEtUgR replied:
Thanks for reviewing! Yes, I thought about that. But extending that thought if the calibration changes in air you might have malicious input e.g. pulling the vehicle in an undesired direction. In any case, I'll drop this check for now and we can add it at least as a warning in air with the stick input workflow changes.

MaEtUgR commented:
For now I removed the validity check for the channel count of every sample matching the calibration parameter. (https://github.com/PX4/PX4-Autopilot/pull/21543#discussion_r1187037528)
And fixed the manual control selector unit tests. I also added a new test that goes through invalid and valid flagged input to verify what changed.

dagar commented on src/modules/rc_update/rc_update.cpp:
```diff
@@ -155,6 +154,12 @@ void RCUpdate::parameters_updated()
 	update_rc_functions();
 
+	_rc_calibrated = _param_rc_chan_cnt.get() > 0
+			 && (_param_rc_map_throttle.get() > 0
```
What about a rover?

What it be enough to start with just RC_CHAN_CNT > 0 and RC_MAP_THROTTLE?

MaEtUgR replied:
I thought about that case. Just one of the 4 main functions needs to be mapped:
channel count > 0 and (throttle or roll or pitch or yaw). Did I do a mistake?

MaEtUgR commented:
I rebased again in the hope Mac CI is more successful. In my eyes this pr is mergeable. Any objections?

MaEtUgR commented:
I rebased again. If no one objects I'll merge this later.
